File: Mozu.SiteBuilder.UX.Admin/Scripts/app/model/ProductTypeAttribute.js
/**
 * @class Taco.model.ProductTypeAttribute
 */
Ext.define('Taco.model.ProductTypeAttribute', {
    extend: 'Taco.core.data.Model',
    fields: [
        { name: 'attributeFQN', type: 'string' },
        { name: 'adminName', type: 'string' },
        { name: 'productTypeId', type: 'auto' },
        { name:'index', type: 'int' },
        { name: 'isLocked', type: 'boolean', defaultValue: false },
        { name: 'isRequired', type: 'boolean', defaultValue: false },
        { name: 'allowMulti', type: 'boolean', defaultValue: false },
        { name: 'isHidden', type: 'boolean', defaultValue: false },
        { name: 'isAdminOnly', type: 'boolean', defaultValue: false },
        { name: 'isProductDetailsOnlyProperty', type: 'boolean', defaultValue: false },
        { name: 'attributeName', type: 'string', persist: false },
        { name: 'allValues', type: 'auto', persist: false, defaultValue: [] },
        { name: 'selectedValues', type: 'auto', defaultValue: [] },
        { name: 'dataType', type: 'string' },
        { name: 'inputType', type: 'string' },
        { name: 'attributeMetadata', type: 'auto', persist: false, defaultValue: [] },
        { name: 'order', type: 'int' }
        
    ],
    getAttributeMetaDataValue:function (key) {
        var val;
        Ext.each(this.get('attributeMetadata'), function (kvp) {
            if (kvp.key == key) {
                val = kvp.value;
            }
        });
        return val;
    },

    idProperty: 'attributeFQN',
    proxy: {
        type: 'ajaxproxy',
        reader: {
            type: 'json',
            root: 'items',
            successProperty: 'success'
        },
        writer: {
            type: 'json',
            allowSingle: false
        }
    }
});

File: Mozu.SiteBuilder.UX.Admin/Api/Models/Attributes/Product/ProductTypeAttribute.cs
using System.Collections.Generic;

namespace Mozu.SiteBuilder.UX.Admin.Api.Models.Attributes.Product
{
    
    public class ProductTypeAttribute
    {
        public string AttributeFQN { get; set; }

        public int? ProductTypeId { get; set; }

        public int? Index { get; set; }

        public bool? IsRequired { get; set; }

        public bool? AllowMulti { get; set; }

        public bool? IsHidden { get; set; }

        public bool? IsLocked { get; set; }

        public List<AttributeValue> SelectedValues { get; set; }

        public List<AttributeValue> AllValues { get; set; }

        public string DataType { get; set; }

        public string InputType { get; set; }

        public string AdminName { get; set; }

        public string AttributeName { get; set; }

        public List<AttributeMetadataItem> AttributeMetadata { get; set; }

        public int Order { get; set; }
    }
}       
